zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/IndexMappingProperty.java

```diff
@@ -7,32 +7,40 @@
  */
 package io.camunda.exporter.schema;
 
+import static java.util.Map.entry;
+
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
-import java.util.HashMap;
+import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Set;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 import org.apache.commons.io.IOUtils;
 
 public record IndexMappingProperty(String name, Object typeDefinition) {
 
-  public static InputStream toPropertiesJson(
-      final Set<IndexMappingProperty> properties, final ObjectMapper mapper) {
-    final var propertiesAsMap =
-        properties.stream()
-            .collect(
-                Collectors.toMap(IndexMappingProperty::name, IndexMappingProperty::typeDefinition));
-    final var propertiesBlock = new HashMap<>();
-    propertiesBlock.put("properties", propertiesAsMap);
-    try {
-      return IOUtils.toInputStream(
-          mapper.writeValueAsString(propertiesBlock), StandardCharsets.UTF_8);
-    } catch (final JsonProcessingException e) {
-      throw new RuntimeException(e);
-    }
+  public static <T> Map<String, T> toPropertiesMap(
+      final Set<IndexMappingProperty> properties,
+      final ObjectMapper mapper,
+      final Function<InputStream, T> typeDeserializer) {
+
+    return properties.stream()
+        .map(
+            prop -> {
+              try {
+                final var typeJson =
+                    IOUtils.toInputStream(
+                        mapper.writeValueAsString(prop.typeDefinition()), StandardCharsets.UTF_8);
+                final var type = typeDeserializer.apply(typeJson);
+                return entry(prop.name(), type);
+              } catch (final JsonProcessingException e) {
+                throw new RuntimeException(e);
```

Comment:
❌ Should we throw an exporter specific Exception, with a proper message? 

❓ How are we handling exceptions in such cases too? this now will fail the schema create or the indexing, right? is that an accepted behavior?

Reply:
This now throws an IllegalStateException given it is a class method of `IndexMappingProperty` it is ok for us to throw an exception as it will crash the exporter because the schema update failed, this is accepted behaviour.

Reply:
Yes, we shall throw an exception if that's expected, but shouldn't we throw `ExporterException` with a proper message then? I just don't like the plain `RuntimeException`.

Reply:
I've changed this now so it's IllegalStateException since it's an instance method on IndexMappingProperty which I think is a more valid and granular than exporterexception